### PR TITLE
Deprecation warning fix

### PR DESCRIPTION
**Pull request by alinioan**
Fixed case where returning strict subclasses of complex in complex doesn't give a deprecation warning.

### CODE REVIEW

**Review by youknowone — CHANGES_REQUESTED** [review]
The change looks good, thank you! Could you check the rustfmt issue? In high chance, it will be fixed by simply running `cargo fmt --all`

**Comment by alinioan** [comment]
I ran the cargo fmt --all comand and made another comit. I hope that fixes the issue

File: vm/src/builtins/complex.rs [hunk 1]
```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
...
+                        "__complex__ returned non-complex (type {}).  \
+                    The ability to return an instance of a strict subclass of complex \
+                    is deprecated, and may be removed in a future version of Python.",
+                        ret_class
+                    ),
+                    1,
+                    vm,
+                )?;
+
+                return Ok(Some((ret.value, true)));
```
Comment (alexandruradovici) [inline vm/src/builtins/complex.rs:78]:
```suggestion
                Ok(Some((ret.value, true)))
```

File: vm/src/builtins/complex.rs [hunk 2]
```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
...
+            } else {
+                return match result.payload::<PyComplex>() {
+                    Some(complex_obj) => Ok(Some((complex_obj.value, true))),
+                    None => Err(vm.new_type_error(format!(
+                        "__complex__ returned non-complex (type '{}')",
+                        result.class().name()
+                    ))),
+                };
```
Comment (alexandruradovici) [inline vm/src/builtins/complex.rs:86]:
```suggestion
                match result.payload::<PyComplex>() {
                    Some(complex_obj) => Ok(Some((complex_obj.value, true))),
                    None => Err(vm.new_type_error(format!(
                        "__complex__ returned non-complex (type '{}')",
                        result.class().name()
                    ))),
                }
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Please fix the conflicts or rebase this.

File: vm/src/builtins/complex.rs [hunk 3]
```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
...
+
+                Ok(Some((ret.value, true)))
+            } else {
+                match result.payload::<PyComplex>() {
+                    Some(complex_obj) => Ok(Some((complex_obj.value, true))),
+                    None => Err(vm.new_type_error(format!(
+                        "__complex__ returned non-complex (type '{}')",
+                        result.class().name()
+                    ))),
+                }
```
Comment (alexandruradovici) [inline vm/src/builtins/complex.rs:86]:
```suggestion
                return match result.payload::<PyComplex>() {
                    Some(complex_obj) => Ok(Some((complex_obj.value, true))),
                    None => Err(vm.new_type_error(format!(
                        "__complex__ returned non-complex (type '{}')",
                        result.class().name()
                    ))),
                };
```

My mistake here, it thought it was the last line in the function.

File: vm/src/builtins/complex.rs [hunk 4]
```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
...
+                        "__complex__ returned non-complex (type {}).  \
+                    The ability to return an instance of a strict subclass of complex \
+                    is deprecated, and may be removed in a future version of Python.",
+                        ret_class
+                    ),
+                    1,
+                    vm,
+                )?;
+
+                Ok(Some((ret.value, true)))
```
Comment (alexandruradovici) [inline vm/src/builtins/complex.rs:78]:
```suggestion
                return Ok(Some((ret.value, true)));
```

My mistake here, it thought it was the last line in the function.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Two small changes due to my mistake.

**Comment by youknowone** [comment]
If you are interested in making commits cleaner, `git rebase` is a good friend for open source contributors.
I am ok with it because I can squash it, but you may want to try for a practice.

File: vm/src/builtins/complex.rs [hunk 5]
```diff
@@ -59,14 +60,31 @@ impl PyObjectRef {
...
+                        "__complex__ returned non-complex (type {}).  \
+                    The ability to return an instance of a strict subclass of complex \
+                    is deprecated, and may be removed in a future version of Python.",
+                        ret_class
+                    ),
+                    1,
+                    vm,
+                )?;
+
+                return Ok(Some((ret.value, true)))
```
Comment (youknowone) [inline vm/src/builtins/complex.rs:78]:
```suggestion
                return Ok(Some((ret.value, true)));
```

**Review by youknowone — APPROVED** [review]
Thank you for contributing!

**Pull request merged**